app/src/main/java/com/example/sun/data/repository/source/ remote/RemoteDataSourceImpl.kt

```diff
@@ -15,17 +14,49 @@ class RemoteDataSourceImpl : CurrentWeatherDataSource.Remote {
         listener: OnResultListener<CurrentWeather>,
         city: String,
     ) {
-        val urlString = Constant.BASE_URL + "weather?q=$city&appid=${Constant.BASE_API_KEY}"
+        val urlString =
+            "${Constant.BASE_URL}${Constant.WEATHER_ENDPOINT}?" +
+                "${Constant.QUERY_PARAM}=$city&" +
+                "${Constant.APPID_PARAM}=${Constant.BASE_API_KEY}&" +
+                "${Constant.UNITS_PARAM}=${Constant.UNITS_VALUE}"
         apiManager.executeApiCall(
             urlString,
+            CurrentWeather::class.java,
             object : OnResultListener<CurrentWeather> {
                 override fun onSuccess(data: CurrentWeather) {
-                    val gson = Gson()
-                    val weather = gson.fromJson(data.toString(), CurrentWeather::class.java)
-                    val date = Date(weather.dt * 1000L)
-                    weather.day = SimpleDateFormat("EEEE yyyy-MM-dd HH:mm:ss", Locale.getDefault()).format(date)
-                    weather.iconWeather = Constant.BASE_ICON_URL + weather.iconWeather + "@2x.png"
-                    listener.onSuccess(weather)
+                    val date = Date(data.dt * 1000L)
+                    data.day = SimpleDateFormat("EEEE yyyy-MM-dd HH:mm:ss", Locale.getDefault()).format(date)
```

Comment:
```suggestion
                    data.day = SimpleDateFormat("EEEE yyyy-MM-dd HH:mm:ss", Locale.getDefault()).format(date)
```
define ``"EEEE yyyy-MM-dd HH:mm:ss"`` là const để tái sử dụng